src/ace.F90

```diff
@@ -512,198 +513,209 @@ end subroutine read_esz
 
-      ! read delayed nu data
-      XSS_index = KNU + 1
-      nuc % nu_d_data = get_real(length)
+          select type(yield => rx % products(1) % yield)
+          type is (Tabulated1D)
+            call yield % from_ace(XSS, KNU + 1)
+          end select
+        end if
 
-      ! =======================================================================
-      ! DELAYED NEUTRON ENERGY DISTRIBUTION
+        KNU = JXS(2) + nint(abs(XSS(JXS(2)))) + 1
+        LNU = nint(XSS(KNU))
+        if (LNU == 1) then
+          ! Polynomial data
+          allocate(Polynomial :: nuc % total_nu)
 
-      ! Allocate space for secondary energy distribution
-      NPCR = NXS(8)
+          ! determine order of polynomial and read coefficients
+          select type (yield => nuc % total_nu)
+          type is (Polynomial)
+            call yield % from_ace(XSS, KNU + 1)
+          end select
+
+        elseif (LNU == 2) then
+          ! Tabulated data
+          allocate(Tabulated1D :: nuc % total_nu)
 
-      ! Check to make sure nuclide does not have more than the maximum number
-      ! of delayed groups
-      if (NPCR > MAX_DELAYED_GROUPS) then
-        call fatal_error("Encountered nuclide with " // trim(to_str(NPCR)) &
-             // " delayed groups while the maximum number of delayed groups &
-             &set in constants.F90 is " // trim(to_str(MAX_DELAYED_GROUPS)))
+          select type(yield => nuc % total_nu)
+          type is (Tabulated1D)
+            call yield % from_ace(XSS, KNU + 1)
+          end select
+        end if
       end if
 
-      nuc % n_precursor = NPCR
-      allocate(nuc % nu_d_edist(NPCR))
+      if (JXS(24) > 0) then
+        ! =======================================================================
+        ! DELAYED NU DATA
+
+        ! Read total yield of delayed neutrons
+        call yield_delayed % from_ace(XSS, JXS(24) + 1)
+
+        idx = JXS(25)
+        total_group_probability = ZERO
+        do i = 1, NPCR
+          ! Set emission mode and decay rate
+          rx % products(1 + i) % emission_mode = EMISSION_DELAYED
+          rx % products(1 + i) % decay_rate = XSS(idx)
+
+          ! Read probability for this precursor group
+          call group_probability % from_ace(XSS, idx + 1)
+
+          ! Set yield based on product of group probability and delayed yield
+          if (all(group_probability % y == group_probability % y(1))) then
+            allocate(Tabulated1D :: rx % products(1 + i) % yield)
+            select type (yield => rx % products(1 + i) % yield)
+            type is (Tabulated1D)
+              yield = yield_delayed
+              yield % y(:) = yield % y(:) * group_probability % y(1)
+              total_group_probability = total_group_probability + group_probability % y(1)
+            end select
+          else
+            call fatal_error("Delayed neutron with energy-dependent group &
+                 &probability not implemented")
```

Comment:
ENDF/B-VII.1 has no energy-dependent delayed group abundances. If they were to occur, I'm not sure we would even be treating them correctly before. Maybe @samuelshaner has further thoughts on this.